File: packages/csharp/HtmlToMarkdown/Metadata/TableExtractionResult.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace HtmlToMarkdown.Metadata;

/// <summary>
/// Result of HTML to Markdown conversion with table extraction.
/// </summary>
public record TableExtractionResult
{
    /// <summary>
    /// The converted Markdown string.
    /// </summary>
    [JsonPropertyName("content")]
    public string Content { get; init; } = string.Empty;

    /// <summary>
    /// Optional extended metadata extracted during conversion.
    /// </summary>
    [JsonPropertyName("metadata")]
    public ExtendedMetadata? Metadata { get; init; }

    /// <summary>
    /// List of tables extracted during conversion.
    /// </summary>
    [JsonPropertyName("tables")]
    public List<TableData> Tables { get; init; } = [];
}

File: packages/java/src/main/java/dev/kreuzberg/htmltomarkdown/TableExtractionResult.java
package dev.kreuzberg.htmltomarkdown;

import com.fasterxml.jackson.annotation.JsonProperty;
import dev.kreuzberg.htmltomarkdown.metadata.HtmlMetadata;
import java.util.List;

/**
 * Result of HTML to Markdown conversion with table extraction.
 *
 * @param content the converted Markdown string
 * @param metadata optional extended metadata (null if not configured)
 * @param tables list of extracted table data
 */
public record TableExtractionResult(
    @JsonProperty("content") String content,
    @JsonProperty("metadata") HtmlMetadata metadata,
    @JsonProperty("tables") List<TableData> tables) {}
